packages/driver/src/util/serialization/log.ts

```diff
@@ -0,0 +1,400 @@
+ */
+export const preprocessLogLikeForSerialization = (props, attemptToSerializeFunctions = false) => {
+  try {
+    if (_.isArray(props)) {
+      return props.map((prop) => {
+        try {
+          return preprocessLogLikeForSerialization(prop, attemptToSerializeFunctions)
+        } catch (e) {
+          return null
+        }
+      })
+    }
+
+    if (_.isPlainObject(props)) {
+      // only attempt to try and serialize dom elements and function (if attemptToSerializeFunctions is set to true)
+      let objWithPossiblySerializableProps = _.pickBy(props, (value) => {
+        const isSerializable = isSerializableInCurrentBrowser(value)
+
+        if (!isSerializable && $dom.isDom(value) || _.isFunction(value) || _.isObject(value)) {
+          return true
+        }
+
+        return false
+      })
+
+      let objWithOnlySerializableProps = _.pickBy(props, (value) => isSerializableInCurrentBrowser(value))
+
+      // assign the properties we know we can serialize here
+      let preprocessed: any = preprocessForSerialization(objWithOnlySerializableProps)
+
+      // and attempt to serialize possibly unserializable props here and fail gracefully if unsuccessful
+      _.forIn(objWithPossiblySerializableProps, (value, key) => {
+        try {
+          preprocessed[key] = preprocessLogLikeForSerialization(value, attemptToSerializeFunctions)
+        } catch (e) {
+          preprocessed[key] = null
+        }
+      })
+
+      return preprocessed
+    }
+
+    if ($dom.isDom(props)) {
+      if (props.length !== undefined && $dom.isJquery(props)) {
+        const serializableArray: any[] = []
+
+        // in the case we are dealing with a jQuery array, preprocess to a native array to nuke any prevObject(s) or unserializable values
+        props.each((key) => serializableArray.push(preprocessLogLikeForSerialization(props[key], attemptToSerializeFunctions)))
+
+        return serializableArray
+      }
+
+      // otherwise, preprocess the element to an object with pertinent DOM properties
+      const serializableDOM = preprocessDomElement(props)
+
+      return serializableDOM
+    }
+
+    /**
+     * When preprocessing a log, there might be certain functions we want to attempt to serialize.
+     * One of these instances is the 'table' key in consoleProps, which has contents that CAN be serialized.
+     * If there are other functions that have serializable contents, the invoker/developer will need to be EXPLICIT
+     * in what needs serialization. Otherwise, functions should NOT be serialized.
+     */
+    if (_.isFunction(props)) {
+      if (attemptToSerializeFunctions) {
+        return {
+          value: preprocessLogLikeForSerialization(props(), attemptToSerializeFunctions),
+          serializationKey: 'function',
+        } as PreprocessedFunction
+      }
+
+      return null
+    }
+
+    return preprocessForSerialization(props)
+  } catch (e) {
+    return null
```

Comment:
I would think so, but the thing I haven't exactly figured out is what would be useful in the debug log. If we do this on every thrown value here, I feel like there could be a lot of noise, but I wonder if we want to have something like "could not serialize property `a` of `b` because it is an instance of `c`". Any ideas what you think might be useful here? I also wonder if we want to have something like namespaced debug names for functions and DOM elements to see the preprocessing/reification in real time?